src/OrbitGl/App.cpp

```diff
@@ -1619,23 +1620,35 @@ orbit_base::Future<ErrorMessageOr<void>> OrbitApp::LoadSymbols(
   return result_future;
 }
 
-orbit_base::Future<ErrorMessageOr<void>> OrbitApp::LoadPresetModule(const std::string& module_path,
-                                                                    const PresetFile& preset_file) {
+ErrorMessageOr<const ModuleData*> OrbitApp::GetLoadedModuleByPath(const std::string& module_path) {
   std::optional<ModuleInMemory> module_in_memory =
       GetTargetProcess()->FindModuleByPath(module_path);
   if (!module_in_memory.has_value()) {
     return ErrorMessage{absl::StrFormat("Module \"%s\" is not loaded by process.", module_path)};
   }
 
-  ModuleData* module_data =
-      module_manager_->GetMutableModuleByPathAndBuildId(module_path, module_in_memory->build_id());
+  const ModuleData* module_data =
+      module_manager_->GetModuleByPathAndBuildId(module_path, module_in_memory->build_id());
   if (module_data == nullptr) {
     ERROR("module \"%s\" was loaded by the process, but is not part of module manager",
           module_path);
     crash_handler_->DumpWithoutCrash();
     return ErrorMessage{"Unexpected error while loading preset."};
   }
 
+  return module_data;
+}
+
+orbit_base::Future<ErrorMessageOr<void>> OrbitApp::LoadPresetModule(const std::string& module_path,
+                                                                    const PresetFile& preset_file) {
+  auto module_data_or_error = GetLoadedModuleByPath(module_path);
+
+  if (module_data_or_error.has_error()) {
+    return module_data_or_error.error();
+  }
```

Comment:
OUTCOME_TRY?

Reply:
the method returns `Future<>` OUTCOME_TRY does not automatically convert it to the correct type.

Reply:
ah interesting!